### DISCUSSION TITLE
How to insert in one-to-one related tables on same dialog

### DISCUSSION THREAD

**Discussion by mgecmez** [serenity-is/Serenity#6771] [Help]
Hello everyone,

I have two one-to-one related tables named Address and CustomerAddress. 

![image](https://github.com/serenity-is/Serenity/assets/16095729/d25eb9fd-8aa0-4d26-94e7-c3a4f8bdccc2)

I don't know if it is possible to have the fields in AddressRow come automatically in CustomerAddressDialog, but for now, I applied a method like below

![image](https://github.com/serenity-is/Serenity/assets/16095729/cab6d748-f165-4f69-a1e5-996df2df6626)
![image](https://github.com/serenity-is/Serenity/assets/16095729/9ff2d3d3-128f-4dc0-98b5-fa0d79e0b077)

Please help if there is a simpler way. But that's not my main problem (maybe it's related to that, I don't know :) ). 

When I try to save it gives the following error:
'Cannot insert the value NULL into column 'AddressId', table 'dbo.CustomerAddress'; column does not allow nulls. INSERT fails.'

How can I solve this problem? If you can help it will save my days!

Thank you for your help in advance.

**Comment by VictorTomaili** [comment]
When you add an expression to a property, the property assumed as foreign field. The handlers did not consider these fields on insert and update. Your error is because the database table has null constraints. And these expression properties are not passed in the query.
